packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -197,31 +197,83 @@ class FileAccessLog extends AccessLog {
 }
 
 /**
- * Represents the properties needed to define a backend
+ * Represents the properties needed to define backend defaults
+ */
+export interface BackendDefaults {
+  /**
+   * Client policy for backend defaults
+   *
+   * @default none
+   */
+  readonly clientPolicy?: ClientPolicy;
+}
+
+/**
+ * Represents the properties needed to define a Virtual Service backend
  */
-export interface Backend {
+export interface VirtualServiceBackendOptions {
   /**
    * The Virtual Service this backend points to
    */
   readonly virtualService: IVirtualService;
 
   /**
-   * Client policy for a backend
+   * Client policy for the backend
    *
    * @default none
    */
   readonly clientPolicy?: ClientPolicy;
 }
 
 /**
- * Represents the properties needed to define backend defaults
+ * Properties for a backend
  */
-export interface BackendDefaults {
+export interface BackendConfig {
   /**
-   * Client policy for backend defaults
-   *
-   * @default none
+   * Config for a Virtual Service backend
    */
-  readonly clientPolicy?: ClientPolicy;
+  readonly virtualServiceBackend: CfnVirtualNode.BackendProperty;
 }
 
+
+/**
+ * Contains static factory methods to create backends
+ */
+export abstract class Backend {
+  /**
+   * Construct a Virtual Service backend
+   */
+  public static virtualServiceBackend(props: VirtualServiceBackendOptions): Backend {
+    return new VirtualServiceBackend(props.virtualService, props.clientPolicy);
+  }
+
+  /**
+   * Return backend config
+   */
+  public abstract bind(_scope: Construct): BackendConfig;
+}
+
+/**
+ * Represents the properties needed to define a Virtual Service backend
+ */
+class VirtualServiceBackend extends Backend {
+
+  constructor (private readonly virtualService: IVirtualService,
+    private readonly clientPolicy: ClientPolicy | undefined) {
```

Comment:
Does this just store these in class properties without having to declare/assign them explicitly?

Reply:
Yep! TypeScript magic 🙂. typescriptlang.org/docs/handbook/2/classes.html#parameter-properties